src/Http/Controllers/Admin/MediaLibraryController.php

```diff
@@ -289,4 +303,21 @@ private function getExtraMetadatas()
             return [$field['name'] => $fieldInRequest];
         });
     }
+
+    /**
+     * @param $sasUrl
+     * @return JsonResponse
+     */
+    public function isValidSas($sasUrl)
```

Comment:
`isValidSas` is technically doing the exact same thing as `policyIsSigned`. Same comment for `isNotValidSas` and `policyIsNotValid`. I agree existing naming doesn't make sense for the Azure implementation because it is not about signing a policy, but we should probably think about renaming to account for both and avoid the code duplication. I think `uploadIsSigned` and `uploadIsNotValid` probably make sense to be more agnostic.

Reply:
@ifox, I agree with you. I did a refactor of `SignS3UploadListener` / `SignAzureUploadListener` to have only one agnostic interface `SignUploadListener`.
Let me know if new implementation is better.

Reply:
Perfect, thank you!